Atlassian sources added (#10739)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/sources/atlassian-jira.md

---
layout: default
title: Atlassian Jira
parent: Sources
grand_parent: Pipelines
nav_order: 7
---

# Atlassian Jira source

You can use the OpenSearch Data Prepper `jira` source to ingest records from one or more [Atlassian Jira](https://www.atlassian.com/software/jira) projects.

## Usage

Set up Jira project access credentials by choosing one of the following options:

- **Basic authentication** (API key authentication): Follow [these instructions](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/).
- **OAuth2 authentication**: Follow [these instructions](https://developer.atlassian.com/cloud/jira/platform/oauth-2-3lo-apps/#faq-rrt-config).

> As an additional optional step, store the credentials in an AWS secret store. If you don't store the credentials in the secret store, then you must provide plain-text credentials directly in the pipeline configuration.

Author: Modified it

### Authentication

You can use one the following authentication methods to access the specified jira host. You must provide one of the following parameters.

| Option   | Required | Type              | Description                                            |
|:---------|:---------|:------------------|:-------------------------------------------------------|
| `basic`  | Yes      | [basic](#basic-authentication)   | Basic Authentication credentials to access a Jira host.  |
| `oauth2` | Yes      | [oauth2](#oauth2-authentication)| OAuth2 Authentication credentials to access a Jira host. |

#### Basic authentication

Either basic or OAuth2 credentials are required to access the Jira site. If you use the `basic` authentication, the following fields are required.

| Option     | Required | Type   | Description                                                                                     |
|:-----------|:---------|:-------|:------------------------------------------------------------------------------------------------|

> | `username` | Yes      | String | A username or reference to secret key storing the username.           |

Reviewer: ```suggestion
| `username` | Yes      | String | A username or reference to the secret key storing the username.           |
```

> | `password` | Yes      | String | A password (API key) or reference to secret key storing the password. |

Reviewer: ```suggestion
| `password` | Yes      | String | A password (API key) or reference to the secret key storing the password. |
```

#### OAuth2 authentication

Either basic or OAuth2 credentials are required to access the jira site. If OAuth2 is used, the following fields are mandatory.

> Either basic or OAuth2 credentials are required to access the Confluence site. If you use OAuth2 is used, the following fields are required.

Reviewer: ```suggestion
Either basic or OAuth2 credentials are required to access the Confluence site. If you use OAuth2, the following fields are required.
```

| Option          | Required | Type   | Description                                                                                     |
|:----------------|:---------|:-------|:------------------------------------------------------------------------------------------------|

> | `client_id`     | Yes      | String | A `client_id` or reference to secret key storing the `client_id`.         |

Reviewer: ```suggestion
| `client_id`     | Yes      | String | A `client_id` or reference to the secret key storing the `client_id`.         |
```

> | `client_secret` | Yes      | String | A `client_secret` or reference to secret key storing the `client_secret`. |

Reviewer: ```suggestion
| `client_secret` | Yes      | String | A `client_secret` or reference to the secret key storing the `client_secret`. |
```

> | `access_token`  | Yes      | String | An `access_token` or reference to secret key storing the `access_token`.   |

Reviewer: ```suggestion
| `access_token`  | Yes      | String | An `access_token` or reference to the secret key storing the `access_token`.   |
```

> | `refresh_token` | Yes      | String | A `refresh_token` or reference to secret key storing the `refresh_token`. |

Reviewer: ```suggestion
| `refresh_token` | Yes      | String | A `refresh_token` or reference to the secret key storing the `refresh_token`. |
```

### Filter

Optionally, you can specify filters to select specific content. If no filters are specified, all the projects and tickets visible for the specified credentials will be extracted and sent to the specified sink in the pipeline.

| Option       | Required | Type   | Description                                    |
|:-------------|:---------|:-------|:-----------------------------------------------|
| `project`    | No       | String | A list of project keys to include or exclude.        |
| `issue_type` | No       | String | A list of issue type filters to include or exclude. |

> | `status`     | No       | String | A list of list of status filters to include or exclude.     |

Reviewer: ```suggestion
| `status`     | No       | String | A list of status filters to include or exclude.     |
```

### AWS secrets

> You can use the following options in the `aws` secrets configuration if you plan to store the credentials in a secret. Storing secrets in AWS secrets store is optional. If secret store is not used, credentials must be specified in the pipeline YAML itself, in plain text.

Reviewer: "AWS secrets store" and "secret store" => AWS Secrets Manager?

Author: modified it

|:---------------|:---------|:-------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `region`       | Yes      | String | The AWS Region to use for credentials. Defaults to the [standard SDK behavior for determining the Region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).                                                                                         |
| `sts_role_arn` | Yes      | String | The AWS Security Token Service (AWS STS) role to assume for requests to Atlassian Jira. Defaults to `null`, which uses the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html). |

> | `secret_id`    | Yes      | Map    | The AWS Secret ARN of the secret that where the credentials are stored.                                                                                                                              

Reviewer: ```suggestion
| `secret_id`    | Yes      | Map    | The Amazon Resource Name (ARN) of the secret where the credentials are stored.                                                                                                                              
```
